### PR TITLE
[HTD-26] Fix: Add `Table of content` component

### PR DESCRIPTION
**Pull request by RBoraurito**
## Pull Request Description

I made a mistake with the commits because I confused the tasks and later on I discovered that I made a mistake with the ticket id

## Assets

| Light | Dark |
| ------ | ----- |
|    image    |   image    |

## Before Merge Checklist

The following steps were checked before merging a PR

- [x] checked and fixed code style issues
- [x] checked and removed unnecessary/junky code
- [x] manually tested introduced solution
- [ ] added needed tests to automate testing this code in the future

### CODE REVIEW

File: src/components/modules/table-content/components/table-content-trigger.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { Link } from '@/components/elements/link';
+import { Text } from '@/components/elements/text';
+
+const TextComponent = ({ children }: Component) => (
+  <Text fontSize="m" fontWeight="bold" color="white">
+    {children}
+  </Text>
+);
+
+export const TableContentTriggerComponent = ({ text, href }: TableContentProps['links'][0]) => {
```
Comment (nephlin7) [inline src/components/modules/table-content/components/table-content-trigger.tsx]: https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html
```suggestion
export const TableContentTriggerComponent = ({ text, href }: TableContentProps['links'][number]) => {
```

File: src/components/modules/table-content/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  StyledTableContentSubitem,
+  StyledTableContentWrapper,
+} from './table-content.styles';
+import type { TableContentProps } from './table-content.types';
+
+import { CaretDown } from '@/components/elements/icon';
+import { Text } from '@/components/elements/text';
+
+const renderAccordion = (link: TableContentProps['links'][0], isCollapsible: boolean) => {
+  const hasSubItems = Array.isArray(link.subItems);
```
Comment (nephlin7) [inline src/components/modules/table-content/index.tsx]:
```suggestion
  const hasSubItems = Array.isArray(link.subItems).length > 0;
```

File: src/components/modules/table-content/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+export const TableContent = ({ isCollapsible, links }: TableContentProps) => {
+  const content = Array.isArray(links) && links.map(link => renderAccordion(link, Boolean(isCollapsible)));
+
+  if (!isCollapsible) return <StyledTableContentWrapper>{content}</StyledTableContentWrapper>;
+
+  return (
+    <StyledTableContentWrapper>
+      <Accordion.Root asChild type="single" defaultValue={links[0].text} collapsible>
+        <>{content}</>
```
Comment (nephlin7) [inline src/components/modules/table-content/index.tsx:51]: Do we need the empty tag?
```suggestion
       {content}
```

**Review by nephlin7 — APPROVED** [review]
@RBoraurito looks good. Check the comments before merge, please.

File: src/components/modules/table-content/index.tsx (hunk 3 above)
Comment (RBoraurito) [reply src/components/modules/table-content/index.tsx:51]: It is because, without that, it gives me an error that it accepts only one child.

**Pull request merged**